### PR TITLE
Add new console command to listen to multiple queues

### PR DESCRIPTION
**Pull request by viktorprogger**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌
| Fixed issues  | #143

### CODE REVIEW

File: src/Command/ListenAllCommand.php [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        foreach ($input->getArgument('channel') as $channel) {
+            $queues[] = $this->queueFactory->get($channel);
+        }
+
+        while ($this->loop->canContinue()) {
+            foreach ($queues as $queue) {
+                $queue->run();
+            }
+
+            sleep($input->getOption('pause'));
```
Comment (xepozz) [inline src/Command/ListenAllCommand.php:69]: I think it's better to use `usleep` and **milliseconds**

File: src/Command/ListenAllCommand.php [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Yiisoft\Yii\Console\ExitCode;
+use Yiisoft\Yii\Queue\Cli\LoopInterface;
+use Yiisoft\Yii\Queue\QueueFactoryInterface;
+
+final class ListenAllCommand extends Command
+{
+    protected static $defaultName = 'queue:listen-all';
+    protected static $defaultDescription = 'Listens the all the given queues and executes messages as they come. ' .
```
Comment (xepozz) [inline src/Command/ListenAllCommand.php:19]: It can be useful not only for development. What about to merge the features to the origin `listen` command?

**Review by xepozz — CHANGES_REQUESTED** [review]

File: src/Command/ListenAllCommand.php (hunk 2 above)
Comment (viktorprogger) [reply src/Command/ListenAllCommand.php:19]: This approach is very CPU-greedy, its usage in production is counterproductive. While most of message brokers allows consumers to listen to them (e.g., socket listening), so they are in idle state until an incoming message, this approach is just an endless loop fressing your CPU.

File: src/Command/ListenAllCommand.php (hunk 1 above)
Comment (viktorprogger) [reply src/Command/ListenAllCommand.php:69]: What is the case for it? This thing shouldn't be used in production, and in dev mode it's ok to wait for one more second IMHO.

File: src/Command/ListenAllCommand.php (hunk 2 above)
Comment (xepozz) [reply src/Command/ListenAllCommand.php:19]: That's up to developers, but I know the company where it was used in production.

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
Looks OK but seems to break all tests.

File: src/Command/ListenAllCommand.php (hunk 2 above)
Comment (viktorprogger) [reply src/Command/ListenAllCommand.php:18]: Yes, that's up to developers. Those who doesn't know what's they doing should be warned.

File: composer.json [hunk 3]
```diff
@@ -70,6 +70,9 @@
         }
     },
     "config": {
+        "platform": {
+            "php": "8.0.2"
+        },
```
Comment (samdark) [inline composer.json:75]: Remove?

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
👍

**Pull request merged**